src/software/gui/robot_diagnostics/threaded_robot_diagnostics_gui.h

```diff
@@ -59,17 +61,19 @@ class ThreadedRobotDiagnosticsGUI : public FirstInFirstOutThreadedObserver<Senso
     void onValueReceived(SensorProto sensor_msg) override;
 
     std::thread run_robot_diagnostics_thread;
+    std::thread run_send_primitives_thread;
     std::shared_ptr<std::promise<void>> termination_promise_ptr;
 
     // Buffers that are shared with the instance of the RobotDiagnosticsGUI so that data
     // can be passed safely
     std::shared_ptr<ThreadSafeBuffer<SensorProto>> sensor_msg_buffer;
-    std::shared_ptr<ThreadSafeBuffer<std::unique_ptr<TbotsProto::PrimitiveSet>>>
-        primitive_buffer;
+    std::shared_ptr<ThreadSafeBuffer<TbotsProto::PrimitiveSet>> primitive_buffer;
 
     // We don't want to miss any updates so we make the buffer larger
     static constexpr std::size_t sensor_msg_buffer_size = 60;
     static constexpr std::size_t primitive_buffer_size  = 60;
+    const unsigned int send_primitive_interval =
```

Comment:
Probably a good idea to suffix the units in the name. Eg. `send_primitive_interval_ms`